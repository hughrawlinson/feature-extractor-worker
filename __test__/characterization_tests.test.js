const { extractFeature, extractFeatureMainThread, availableFeatureExtractors } = require("../src/main/");
const { extractFeatures, extractFeaturesObservable } = require('../src/main/feature-extractor');

window.AudioContext = jest.fn().mockImplementation(() => ({
    decodeAudioData: jest.fn().mockImplementation(() => ({
      numberOfChannels: 3,
      // TODO: Test a length with a power of two
      getChannelData: jest.fn().mockImplementation(() => new Float32Array(19932))
    }))
}));

const mockAudioBlob = {
  arrayBuffer: jest.fn().mockImplementation(() => ({}))
};

describe('extractFeatures', () => {
  it('does something', async function() {
    const call = await extractFeature({
      audioBlob: mockAudioBlob,
      audioFeatures: ['amplitudeSpectrum'],
      extractionParams: { bufferSize: 20 }
    });
    expect(call).toMatchSnapshot();
  });

  it('does something on the main thread', async function() {
    expect(await extractFeatureMainThread({
      audioBlob: mockAudioBlob,
      audioFeatures: ['amplitudeSpectrum'],
      extractionParams: { bufferSize: 1024 }
    }))
  });

  it('makes certain features available', () => {
    expect(availableFeatureExtractors).toMatchSnapshot();
  });
});

describe('feature-extractor', () => {
  it('is callable', async function() {
    expect(await extractFeatures([new Float32Array(15)], ['zcr'], {
      bufferSize: 8,
      hopSize: 0,
      zeroPadding: 0,
      windowingFunction: 0,
    })).toMatchSnapshot();
  });
});

describe('feature-extractor-observable', () => {
  it('is callable', async function() {
    extractFeaturesObservable(
      [],
      ['zcr'],
      { bufferSize: 32 },
      { batchSize: 10 }
    ).subscribe(result => {
      console.log(result);
      return expect(result).toMatchInlineSnapshot();
    });
  });
});
