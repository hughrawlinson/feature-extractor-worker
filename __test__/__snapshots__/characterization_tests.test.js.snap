// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractFeatures does something 1`] = `"MOCK_WORKER_POSTMESSAGE_DATA"`;

exports[`extractFeatures makes certain features available 1`] = `
Array [
  "rms",
  "energy",
  "spectralSlope",
  "spectralCentroid",
  "spectralRolloff",
  "spectralFlatness",
  "spectralSpread",
  "spectralSkewness",
  "spectralKurtosis",
  "zcr",
  "loudness",
  "perceptualSpread",
  "perceptualSharpness",
  "mfcc",
  "chroma",
  "powerSpectrum",
  "spectralFlux",
  "buffer",
  "complexSpectrum",
  "amplitudeSpectrum",
]
`;

exports[`feature-extractor is callable 1`] = `
Array [
  Array [
    Object {
      "zcr": 0,
    },
    Object {
      "zcr": 0,
    },
  ],
]
`;
